permits/views.py

```diff
@@ -803,6 +810,152 @@ def get(self, request, path, *args, **kwargs):
+
+        if not archive:
+            raise SuspiciousOperation
+
+        if not self.request.user == archive.archivist:
+            messages.error(self.request, self.error_message)
+            return redirect(self.success_url)
+
+        return super(ArchivedPermitRequestDeleteView, self).post(
+            request, *args, **kwargs
+        )
+
+    def get_success_url(self):
+        messages.success(self.request, self.success_message % self.object.pk)
+        return self.success_url
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(check_mandatory_2FA, name="dispatch")
+class ArchivedPermitRequestDownloadView(View):
+    error_message = _("Vous n'avez pas les permissions pour télécharger cette archive")
+
+    def get(self, request, *args, **kwargs):
+        archive = models.ArchivedPermitRequest.objects.filter(
+            permit_request=kwargs.get("pk")
+        ).first()
+        if not archive:
+            raise SuspiciousOperation
+
+        if not (
+            self.request.user.groups.all() & archive.archivist.groups.all()
+        ).exists():
+            messages.error(request, self.error_message)
+            return redirect(reverse_lazy("permits:archived_permit_request_list"))
+
+        zippath = services.compress_directory(archive.path)
+        response = FileResponse(open(zippath, "rb"))
+        os.remove(zippath)
+        return response
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(check_mandatory_2FA, name="dispatch")
+class ArchivedPermitRequestBulkDownloadView(View):
+    error_message = _("Vous n'avez pas les permissions pour télécharger ces archives")
+    info_message = _("Rien à télécharger")
+
+    def get(self, request, *args, **kwargs):
+        to_download = self.request.GET.getlist("to_download")
+        if not to_download:
+            messages.info(request, self.info_message)
+            return redirect(reverse_lazy("permits:archived_permit_request_list"))
+
+        archives = []
+        for archive_id in to_download:
+            archive = models.ArchivedPermitRequest.objects.filter(
+                permit_request=archive_id
+            ).first()
+            if not archive:
+                raise SuspiciousOperation
+
+            if not (
+                self.request.user.groups.all() & archive.archivist.groups.all()
+            ).exists():
+                messages.error(request, self.error_message)
+                return redirect(reverse_lazy("permits:archived_permit_request_list"))
+
+            archives.append(archive)
+
+        parent_name = f"Archive_{datetime.today().strftime('%d.%m.%Y.%H.%M.%S')}"
+        parent_path = os.path.join(settings.ARCHIVE_ROOT, parent_name)
+        os.mkdir(parent_path)
+
+        for archive in archives:
+            shutil.copytree(
+                src=archive.path, dst=os.path.join(parent_path, archive.dirname)
+            )
+
+        zippath = services.compress_directory(parent_path)
```

Comment:
Can this fail, and thus throw an exception ? It probably should, in case of failure
In this case, try and catch the exception here and return an appropriate error HTTP response.